### DISCUSSION TITLE
How to allows users to add custom domains to a specific page?

### DISCUSSION THREAD

**Discussion by jessie-mills-dev** [vercel/community#5655] [Help]
### Summary

Hey, I'm trying to wrap my head around how to do this. I have a multi-tenant React JS app that uses Supabase for the backend, and Vercel for deployment. 

Each tenant has as roadmap page that is public, and is accessible by a url, eg: mypublicurl.co/2901481048109481094. However for my pro plan I want my users to be able to replace this url with a custom domain, eg roadmap.johnnyscoolsite.com

I've set up an input etc so that in my db I'm collecting their custom URLs, but am struggling with ensuring they redirect to the unique url that each user has for the roadmap.

eg say as Johnny's cool site I go to my DNS settings - how do I redirect to the specific url?? I've been looking at dynamic redirects and wildcard domains but I feel like it should be simpler since the URL already exists... just needs to cover it.

### Steps to Reproduce

Not really something to reproduce, looking for general advice.

**Comment by aap262** [comment]
Hey, have you already found out? I'm looking for this information for weeks and it would be a blessing if you could help me out!

**Comment by amyegan** [comment]
There's more than one way you could do this, and the right one for you will depend on your app structure and what you're comfortable doing. We have a few templates that do versions of what you're asking, and I recommend looking at these to help you decide.

- [Hostname Rewrites using Edge Middleware](https://vercel.com/templates/next.js/hostname-rewrites)
- [Platforms Starter Kit](https://vercel.com/templates/next.js/platforms-starter-kit)
- [Static.fun](https://vercel.com/templates/next.js/static-fun-wildcard-domains)

**Comment by versecafe** [comment]
You can also check out
- [Vercel Domains API](https://vercel.com/templates/next.js/domains-api)
- [Dub custom domain link redirects](https://github.com/dubinc/dub/blob/cf9dfa01a44997434f4fdb53fa56ac0aa8796c96/apps/web/lib/api/domains.ts#L57)
